app/models/manager_refresh/inventory_collection/builder.rb

```diff
@@ -0,0 +1,234 @@
+module ManagerRefresh
+  class InventoryCollection
+    class Builder
+      class MissingModelClassError < StandardError; end
+
+      require_nested :CloudManager
+      require_nested :NetworkManager
+      require_nested :StorageManager
+      require_nested :InfraManager
+
+      include ::ManagerRefresh::InventoryCollection::Builder::Shared
+
+      # Default options for builder
+      #   :auto_object_attributes
+      #     - auto creates inventory_object_attributes from target model_class setters
+      #     - attributes used in InventoryObject.add_attributes
+      #   :auto_model_class
+      #     - tries to set model_class from persister class
+      #     - @see method auto_model_class
+      #   :without_model_class
+      #     - if false and no model_class derived or specified, throws exception
+      #   :adv_settings_enabled
+      #     - values from Advanced settings (doesn't overwrite values specified in code)
+      #     - see persister options() method
+      def self.default_options
+        {
+          :auto_object_attributes => true,
+          :auto_model_class       => true,
+          :without_model_class    => false,
+          :adv_settings_enabled   => true
+        }
+      end
+
+      # Entry point
+      # Creates builder and builds data for inventory collection
+      # @param name [Symbol] InventoryCollection.association value
+      #        (optional) method with this name also used for concrete inventory collection specific properties
+      # @param persister_class [Class] used for "guessing" model_class
+      # @param options [Hash]
+      def self.prepare_data(name, persister_class, options = {})
+        options = default_options.merge(options)
+        builder = new(name, persister_class, options)
+        builder.construct_data
+
+        yield(builder) if block_given?
+
+        builder
+      end
+
+      # @see prepare_data()
+      def initialize(name, persister_class, options = self.class.default_options)
```

Comment:
hm, might be cleaner to use kwargs?

so e.g. 
```ruby
def initialize(name, persister_class, auto_object_attributes: true, auto_model_class:true, without_model_class:false, adv_settings_enabled: true)
```
?

Reply:
what does it mean - 'kwargs'? 
I'm not sure how this example works, what is name of this hash?
And why is it better if this default options are shared across initialize(), prepare_data (class method) and (in extended form) in persister's add_collection() method?

Reply:
kwargs == key word arguments

So method can take optional key word parameters, with default value, if you don't send them (from ruby 2.0). It's cleaner than just passing hash, since you can then document all parameter in the standard way.

From what I see, if we will use it as kwargs, we do not need the default method?

Reply:
Through this options, you can assign more than these 4 values (specifically adv_settings and shared_options). 
If you use kwargs there, do you mean that you should use kwargs in ```prepare_data``` method? Which leads to two documentations of the same...and repeating the same code on more places